install_files/ansible-base/roles/restore/tasks/perform_restore.yml

```diff
@@ -23,8 +23,10 @@
   connection: local
   become: no
   unarchive:
+    # Avoid creating an unnecessary local copy
+    remote_src: yes
     dest: "{{ torrc_check_dir.path }}/backup/"
-    src: "{{ restore_file }}"
+    src: "{{playbook_dir}}/{{ restore_file }}"
```

Comment:
For `.tar.gz` files, the `unarchive` module appears to expect full paths and will fail (misleadingly) with "Failed to find handler" on relative paths; see https://github.com/ansible/ansible/issues/35645. The "Failed to find handler" issue arises only if we enable the `remote_src` setting, which suggests to me that otherwise, Ansible may pass along the full path to the copy it has created.